crates/rpc-types/src/genesis.rs

```diff
@@ -27,3 +79,172 @@ pub struct OptimismBaseFeeInfo {
     /// EIP-1559 denominator after canyon
     pub eip1559_denominator_canyon: Option<u64>,
 }
+
+impl OptimismBaseFeeInfo {
+    /// Extract the Optimism base fee info from `OtherFields` object.
```

Comment:
this should document this will check for an `optimism` object
and is intended to be parsed from the genesis object